### ISSUE TITLE
Null value instead of 0?

### ISSUE THREAD

**Issue by acirinelli** [vuejs-tips/v-money#15]
Is there a way you can add an option of setting the value to null when you focus on the input? The 0 alone is causing multiple issues for our users. In some cases the cursor is in front of the 0 which isn't helpful, and in others, they try to delete the zero before typing.

Basically, if the value is 0 on focus, the input numbers should clear. If the value is null on blur, it should go back to 0.

**Comment by neves** [comment]
Can you record the issue while playing on this page: https://vuejs-tips.github.io/v-money/
You can use https://screencast-o-matic.com/

**Comment by acirinelli** [comment]
Here is a video clip:
edit: [video deleted from dropbox]

**Comment by acirinelli** [comment]
@neves if you don't want to update your package with this idea, no worries. I did find a jquery package that does what I'm looking for. Reference: https://github.com/autoNumeric/autoNumeric

**Comment by neves** [comment]
Now I see. That's really annoying, so I'm marking it to be fixed.

**Comment by discovermotion** [comment]
Here's my wish - if the (value == null) the input control should not display "$0" it should be displayed as empty initially.  If the (value == 0 or other valid number) it should display the formatted value. Thanks for your work on this.

**Comment by discovermotion** [comment]
Any status on an update for this?  Thanks.

**Comment by renatosistemasvc** [comment]
I want you to provide the user with the option to leave the field blank

**Comment by gfviegas** [comment]
@neves any progress?

**Comment by liuyao61** [comment]
Is there any fix for this?

**Comment by acirinelli** [comment]
I gave up and moved on.. For those still waiting/hoping for this fix, the [vue-autoNumeric](https://github.com/autoNumeric/vue-autoNumeric) package is really nice and behaves as one would expect!

**Comment by DinsmoreDesign** [comment]
@acirinelli Good looking out! I've used v-money for several projects in the past, but the ability to set the value to blank before any user input is a requirement of my latest project and it doesn't look like it's possible right now. I'll definitely give this a whirl, instead!

**Comment by renatosistemasvc** [comment]
I downloaded the project and made some changes. Now it works with blank value.
I do not know why the project maintainer does not change.
I'll post the changes here for anyone who wants to change

1- You download the project and copy the src folder somewhere inside your project;
2 - Inside your main.js file you initialize the local library:

```
import money from 'new-path-local'
Vue.use(money, {precision: 4})
```

3 - in the component.vue file (inside the src folder) change the rows:

of
```
value: {
      required: true,
      type: [Number, String],
      default: 0
},

```
for
```
value: {
      required: true,
      default: 0
    },
```
___________________________________________
of
```
watch: {
.....
}
```

for
```
watch: {
    value: {
      immediate: true,
      handler (newValue, oldValue) {

        if(!newValue){
          var formatted = '';
        }else{
          var formatted = format(newValue, this.$props)
        }
       
        if (formatted !== this.formattedValue) {
          this.formattedValue = formatted
        }
      }
    }
  },
```
___________________________________________
of
```
change (evt) {
.....
}
```
for
```
change (evt) {
      this.$emit('input', this.masked ? evt.target.value : ( (!evt.target.value) ? '' : unformat(evt.target.value, this.precision)))
    }
```
4 - in the directive.js file (inside the src folder) change the rows:

of
```
el.oninput = function () {
......
}
```
for
```
el.oninput = function () {
    var positionFromEnd = el.value.length - el.selectionEnd
    
    if(el.value == 'R$ 0,0' || !el.value){
      el.value = ''
    }else{
      el.value = format(el.value, opt)
    }

    positionFromEnd = Math.max(positionFromEnd, opt.suffix.length) // right
    positionFromEnd = el.value.length - positionFromEnd
    positionFromEnd = Math.max(positionFromEnd, opt.prefix.length + 1) // left
    setCursor(el, positionFromEnd)
    el.dispatchEvent(event('change')) // v-model.lazy
}
```
5 - in the index.js file (inside the src folder) remove the row:

`const VERSION = proccess.env.VERSION`

With these modifications, the project works well with blank values.

**Comment by dbomfim** [comment]
@renatosistemasvc Which Vue version did you used to compile?! Mine it's not working... 

  Unknown option --config.

**Comment by paulodiogo** [comment]
Is there any fix for this?

**Comment by Jared-Dahlke** [comment]
> I gave up and moved on.. For those still waiting/hoping for this fix, the [vue-autoNumeric](https://github.com/autoNumeric/vue-autoNumeric) package is really nice and behaves as one would expect!

agreed. vue-autoNumeric solves 2 major problems for me that v-money does not solve:
1. it doesn't force the user to type in decimals
2.it shows as blank if the value is null

**Comment by kalibano** [comment]
any updates regarding fix?

**Comment by SharpBCD** [comment]
Still nothing? Come on... it is open for 2+ years. Can't be that hard. :-(

**Comment by WilliamColmenares** [comment]
it looks this package is no longer maintained, thumbs up for vue-autonumeric.

**Comment by ggirardi1** [comment]
Hello friends!

Passing by to share with you the way I cut this problem.

Basically I wrapped the original [jonathanpmartins/v-money3](https://github.com/jonathanpmartins/v-money3) component inside its own component, handling the value before issuing the `update:modelValue` event.

It works perfectly for me.

Hope this helps!

```vue
<template>
    <money3
        class="border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm"
        :modelValue="modelValue"
        @change="change"
        v-bind="opt"
    ></money3>

</template>

<script>
    import { Money3Component } from 'v-money3'
    import { unformat } from 'v-money3';

    export default {
        data() {
            return {
                opt: {
                    masked: false,
                    prefix: 'R$ ',
                    suffix: '',
                    thousands: '.',
                    decimal: ',',
                    precision: 2,
                    allowBlank: true,
                }
            }
        },

        name: 'CurrencyInput',

        props: ['modelValue','options'],

        emits: ['update:modelValue'],

        methods: {
            change(event) {
                let unformatted = parseFloat(unformat(event.target.value, this.opt));
                if (this.opt.allowBlank)
                    unformatted = unformatted == 0 ? null : unformatted;
                
                this.$emit('update:modelValue', unformatted);
            }
        },

        created() {
            this.opt = Object.assign({}, this.opt, this.options);
        },

        components: {
            money3: Money3Component
        }
    }
</script>
```

**Comment by andreLuis1506** [comment]
> @renatosistemasvc Which Vue version did you used to compile?! Mine it's not working...
> 
> Unknown option --config.

this is happening to me too, any solution?
